src/mame/capcom/gunsmoke.cpp

```diff
@@ -945,18 +993,19 @@ ROM_END
-GAME( 1985, gunsmokej,  gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom",                   "Gun.Smoke (Japan, 851115)",           MACHINE_SUPPORTS_SAVE ) // GSJ_03
-GAME( 1986, gunsmokeu,  gunsmoke, gunsmoke, gunsmokeu, gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (US, 860408)",              MACHINE_SUPPORTS_SAVE ) // GSA_03
-GAME( 1985, gunsmokeua, gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (US, 851115, set 1)",       MACHINE_SUPPORTS_SAVE ) // GSR_03 (03A on the chip)
-GAME( 1986, gunsmokeub, gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (US, 851115, set 2)",       MACHINE_SUPPORTS_SAVE ) // GSR_03
-GAME( 1985, gunsmokeg,  gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom",                   "Gun.Smoke (Germany, censored)",       MACHINE_SUPPORTS_SAVE ) // Has 'World' regional warning, but game censored for German market
+GAME( 1985, gunsmoke,   0,        gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom",                   "Gun.Smoke (World, 1985/11/15)",                 MACHINE_SUPPORTS_SAVE ) // GSE_03
+GAME( 1985, gunsmokeb,  gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "bootleg",                  "Gun.Smoke (World, 1985/11/15) (bootleg)",       MACHINE_SUPPORTS_SAVE ) // based on above version, warning message patched out
+GAME( 1985, gunsmokej,  gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom",                   "Gun.Smoke (Japan, 1985/11/15)",                 MACHINE_SUPPORTS_SAVE ) // GSJ_03
+GAME( 1986, gunsmokeu,  gunsmoke, gunsmoke, gunsmokeu, gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (USA and Canada, 1986/04/08)",        MACHINE_SUPPORTS_SAVE ) // GSA_03
+GAME( 1985, gunsmokeua, gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (USA and Canada, 1986/01/20)",        MACHINE_SUPPORTS_SAVE ) // GSA_03Y (GSR 03Y on the chip)
+GAME( 1985, gunsmokeub, gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (USA and Canada, 1985/11/15, set 1)", MACHINE_SUPPORTS_SAVE ) // GSR_03 (03A on the chip)
+GAME( 1985, gunsmokeuc, gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom (Romstar license)", "Gun.Smoke (USA and Canada, 1985/11/15, set 2)", MACHINE_SUPPORTS_SAVE ) // GSR_03
+GAME( 1985, gunsmokeg,  gunsmoke, gunsmoke, gunsmoke,  gunsmoke_state, empty_init, ROT270, "Capcom",                   "Gun.Smoke (Germany, 1985/11/15, censored)",     MACHINE_SUPPORTS_SAVE ) // GSG_03, has 'World' regional warning, but game censored for German market
```

Comment:
Where do these dates come from and what format did Capcom use?  Capcom generally used YYMMDD date codes (two-digit year without century, no punctuation), as famously displayed on Street Fighter boot screens.  We date/version strings as-is rather than messing with the formatting, so this looks like an undesirable change.

Reply:
They're stored as 'yyyymmdd' on the main program ROM (the metadata starts at 0x7E50). I've switched to 'yyyymmdd'.

Thanks!!

Reply:
No they aren’t.  They have no century – they’re just like “851113”.  I checked it now.  Please put it back how it was with the date code quoted verbatim from the ROM.  Arbitrarily adding the century is only going to make things confusing.

Reply:
![image](https://user-images.githubusercontent.com/5889052/208507499-ab9b96f7-6fca-41df-a557-06898fe1e620.png)

Nop! It's yyymmdd, see 0x7E66-0x7E66. Also, it's documented on the driver, see comment at line 999.

Reply:
Oh my bad, I looked in the wrong location, sorry.

Reply:
Thanks for reviewing!